src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.cs

```diff
@@ -577,7 +594,65 @@ static void TestMain()
 True
 ";
 
-            CompileAndVerify(source + InstrumentationHelperSource, emitOptions: EmitOptions.Default.WithInstrument("Test.Flag"), expectedOutput: expectedOutput);
+            CompileAndVerify(source + InstrumentationHelperSource + XunitFactAttributeSource, emitOptions: EmitOptions.Default.WithInstrument("Test.Flag"), expectedOutput: expectedOutput);
+        }
+
+        [Fact]
+        public void XunitFactAttribute()
+        {
+            string source = @"
+using System;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        MethodWithFactAttribute(1);
+        MethodWithOutFactAttribute(1);
+        MethodWithFactAttribute(1);
+    }
+
+    [Xunit.Fact]
+    static int MethodWithFactAttribute(int x)
+    {
+        var y = x + 1;
+        var z = y + 1;
+        return z;
+    }
+
+    static int MethodWithOutFactAttribute(int x)
```

Comment:
It takes some cleverness to be able to tell from the test output which method is doing the flushing. Consider checking the generated IL (see an example above) or calling the method that does the flushing twice.